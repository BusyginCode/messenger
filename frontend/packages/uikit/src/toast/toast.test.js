import React from 'react'
import { shallow } from 'enzyme'
import Toast from './toast'

const icon = <svg />

describe('uikit/Toast', () => {
  it('renders content as a text with success intent', () => {
    const toast = shallow(<Toast icon={icon} content="Simple text" onClose={jest.fn()} intent="success" />)

    expect(toast).toMatchSnapshot()
  })

  it('renders toast with warning intent', () => {
    const onClose = jest.fn()
    const toast = shallow(<Toast icon={icon} content="Simple text" onClose={onClose} intent="warning" />)

    toast
      .find('Svg(close)')
      .at(0)
      .prop('onClick')()

    expect(onClose).toHaveBeenCalledTimes(1)
    expect(toast).toMatchSnapshot()
  })
})
