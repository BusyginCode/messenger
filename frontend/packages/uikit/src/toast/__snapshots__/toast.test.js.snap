// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`uikit/Toast renders content as a text with success intent 1`] = `
<div
  className="toast success"
  data-role="toast-success"
>
  <div
    className="fn-iconContainer iconContainer"
  >
    <svg />
  </div>
  <div
    className="content"
  >
    Simple text
  </div>
  <Svg(close)
    className="close"
    onClick={[Function]}
  />
</div>
`;

exports[`uikit/Toast renders toast with warning intent 1`] = `
<div
  className="toast warning"
  data-role="toast-warning"
>
  <div
    className="fn-iconContainer iconContainer"
  >
    <svg />
  </div>
  <div
    className="content"
  >
    Simple text
  </div>
  <Svg(close)
    className="close"
    onClick={[Function]}
  />
</div>
`;
